module/data/actor/templates/common.mjs

```diff
@@ -0,0 +1,67 @@
+import SystemDataModel from "../../abstract.mjs";
+import { MappingField } from "../../fields.mjs";
+import CurrencyTemplate from "../../shared/currency.mjs";
+import AbilityData from "../ability.mjs";
+
+/**
+ * A template for all actors that share the common template.
+ *
+ * @property {Object<string, AbilityData>} abilities  Actor's abilities.
+ * @mixin
+ */
+export default class CommonTemplate extends SystemDataModel.mixin(CurrencyTemplate) {
+
+  /** @inheritdoc */
+  static defineSchema() {
+    return this.mergeSchema(super.defineSchema(), {
+      abilities: new MappingField(new foundry.data.fields.EmbeddedDataField(AbilityData), {
+        initialKeys: CONFIG.DND5E.abilities
+      }, {label: "DND5E.Abilities"})
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  static migrateData(source) {
+    this.#migrateACData(source);
+    this.#migrateMovementData(source);
```

Comment:
To avoid issues with subclasses not being able to access this private migration you should do:
```suggestion
    CommonTemplate.#migrateACData(source);
    CommonTemplate.#migrateMovementData(source);
```